app/helpers/foreman_host_reports/reports_dashboard_helper.rb

```diff
@@ -0,0 +1,74 @@
+module ForemanHostReports
+  module ReportsDashboardHelper
+    def host_reports_searchable_links(name, search, counter, format)
+      content_tag :li, :style => "margin-bottom: 5px" do
+        content_tag(:span, sanitize('&nbsp;'), :class => 'label', :style => "background-color: #{host_reports_report_color[counter]}") +
+          sanitize('&nbsp;') +
+          link_to(name, hosts_path(:search => search), :class => "dashboard-links") +
+          content_tag(:span, send(counter, format), class: 'pull-right')
+      end
+    end
+
+    def host_reports_report_color
+      {
+        :change_hosts => "#4572A7",
+        :failure_hosts => "#AA4643",
+        :nochange_hosts => "#DB843D",
+        :disabled_hosts => "#92A8CD",
+      }
+    end
+
+    def host_reports_get_overview(options = {})
+      format = options[:format]
+      state_labels = {
+        change_hosts: _('Change'),
+        failure_hosts: _('Failure'),
+        nochange_hosts: _('No Change'),
+        disabled_hosts: _('Disabled alerts'),
+      }
+
+      {
+        data: state_labels.map { |key, label| [label, send(key, format), host_reports_report_color[key]] },
+        searchUrl: hosts_path(search: '~VAL~'),
+        searchFilters: state_labels.each_with_object({}) do |(key, filter), filters|
+          filters[filter] = send(key, format)
+        end,
+      }
+    end
+
+    def latest_reports
+      hosts = Host.authorized(:view_hosts, Host)
+      HostReport.authorized(:view_host_reports).my_reports
+                .where(host_id: hosts.pluck(:id))
+                .reorder(:last_report => :desc)
+                .limit(9)
+                .preload(:host)
```

Comment:
I have realized that the my_reports scope actually uses a helper to join host authorized host reports:

```ruby
  scope :my_reports, lambda {
    if !User.current.admin? || Organization.expand(Organization.current).present? || Location.expand(Location.current).present?
      joins_authorized(Host, :view_hosts)
    end
  }
```

Could you simplify the query to just:

```ruby
      HostReport.authorized(:view_host_reports).my_reports
                .reorder(:last_report => :desc)
                .limit(9)
                .preload(:host)
```

removing the where condition and the helper? Then try this with hosts belonging to different organizations (A,B) and try it under regular user (non-admin) from org A. You should not see hosts from org B.